src/dispatching/dialogue/storage/trace_storage.rs

```diff
@@ -0,0 +1,55 @@
+use std::{fmt::Debug, sync::Arc};
+
+use futures::future::BoxFuture;
+use log::trace;
+
+use super::Storage;
+
+/// Storage wrapper for logging purposes
+///
+/// Reports about any dialogue update or removal action on `trace` level
+pub struct TraceStorage<S> {
+    inner: Arc<S>,
```

Comment:
What's the point of this `Arc<_>`?

Reply:
Currently, we are pretty much limited to `Arc<S>` due to `Arc<Self>` requirement in
https://docs.rs/teloxide/0.3.3/teloxide/dispatching/dialogue/trait.Storage.html#tymethod.remove_dialogue
https://docs.rs/teloxide/0.3.3/teloxide/dispatching/dialogue/trait.Storage.html#tymethod.update_dialogue
It's possible to create `Arc` on every new request though, but that would result in unnecessary allocations

Reply:
Huh, that's unfortunate. Methods should probably accept `&Self` — `Arc` derefs anyway